### PR TITLE
Updates RSPEC and removes S1128 and S2327 rules from sonarway profile

### LINKED ISSUE #3836: Remove S1128 and S2327 from the default Quality Profile (SonarWay)
S1128 is quite nice, but its performance needs to be improved in order to be in SonarWay ( #3761)
S2327 is controversial (see #2490)

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**
Fixes #3836

### CODE REVIEW

File: analyzers/rspec/cs/S1313_c#.html [hunk 1]
```diff
@@ -33,14 +36,18 @@ <h2>Compliant Solution</h2>
 var address = IPAddress.Parse(ip);
 </pre>
 <h2>Exceptions</h2>
+<p>No issue is reported for the following cases because they are not considered sensitive:</p>
```
Comment (andrei-epure-sonarsource) [inline analyzers/rspec/cs/S1313_c#.html:39]: This isn't implemented yet (see #3450) - we've had to manually remove the change related to S1313 from the previous releases (you can revert the S1313_* files to the master version)

File: analyzers/rspec/cs/S5042_c#.html [hunk 2]
```diff
@@ -1,53 +1,87 @@
...
+<h2>Compliant Solution</h2>
+<pre>
+int THRESHOLD_ENTRIES = 10000;
+int THRESHOLD_SIZE = 1000000000; // 1 GB
+double THRESHOLD_RATIO = 10;
+int totalSizeArchive = 0;
+int totalEntryArchive = 0;
+
+FileStream zipToOpen = new FileStream(@"ZipBomb.zip", FileMode.Open);
+ZipArchive archive = new ZipArchive(zipToOpen, ZipArchiveMode.Read);
```
Comment (andrei-epure-sonarsource) [inline analyzers/rspec/cs/S5042_c#.html:40]: I'm not sure that we can raise issues on this one. We have #3621 to deal with this rule in the upcoming sprint - please revert the changes to S5042_* files just to avoid having smth in the docs which is not in the implementation

File: analyzers/rspec/cs/S5547_c#.html [hunk 3]
```diff
@@ -1,24 +1,38 @@
 <p><a href="https://en.wikipedia.org/wiki/Strong_cryptography">Strong cipher algorithms</a> are cryptographic systems resistant to cryptanalysis, they
-are not vulnerable to well-known attacks like brute force attacks for example. </p>
-<p>It is recommended to use only cipher algorithms intensively tested and promoted by the cryptographic community.</p>
+are not vulnerable to well-known attacks like brute force attacks for example.</p>
```
Comment (andrei-epure-sonarsource) [inline analyzers/rspec/cs/S5547_c#.html:2]: Please revert the S5547_* changes - we have #3330 to fix first (we'll deal with it in the upcoming sprint)

We've been reverting them manually in the past releases as well (see notes in #3559 and #3607)

File: analyzers/rspec/cs/S5659_c#.html [hunk 4]
```diff
@@ -1,9 +1,7 @@
 <p>If a JSON Web Token (JWT) is not signed with a strong cipher algorithm (or not signed at all) an attacker can forge it and impersonate user
 identities.</p>
-<ul>
-  <li> Don't use <code>none</code> algorithm to sign or verify the validity of an algorithm. </li>
-  <li> Don't use a token without verifying its signature before. </li>
-</ul>
+<p> * Don't use <code>none</code> algorithm to sign or verify the validity of a token.</p>
```
Comment (andrei-epure-sonarsource) [inline analyzers/rspec/cs/S5659_c#.html:3]: please run the rspec script again for S5659 - there's a glitch in the rule-api when there is a ` *` in the JIRA instead of `*` - I removed the space from the RSPEC and rerunning it should correct it

File: analyzers/src/SonarAnalyzer.CSharp/RspecStrings.resx [hunk 5]
```diff
@@ -12425,7 +12425,7 @@
     <value>Critical Security Hotspot</value>
   </data>
   <data name="S5042_Description" xml:space="preserve">
-    <value>Expanding archive files is security-sensitive. For example, expanding archive files has led in the past to the following vulnerabilities:</value>
+    <value>Successful Zip Bomb attacks occur when an application expands untrusted archive files without controlling the size of the expanded data, which can lead to denial of service. A Zip bomb is usually a malicious archive file of a few kilobytes of compressed data but turned into gigabytes of uncompressed data. To achieve this extreme compression ratio, attackers will compress irrelevant data (eg: a long string of repeated bytes). </value>
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/RspecStrings.resx:12428]: (this is fine, it's just an improved phrasing of the rule)

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
I've left a few necessary changes - they apply for both C# and VB .NET

(it's about RSPECs that have been updated and we didn't update the rules yet to synchronize with the RSPECs)

File: analyzers/src/SonarAnalyzer.CSharp/RspecStrings.resx (hunk 5 above)
Comment (csaba-sagi-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/RspecStrings.resx:12428]: So I should not revert it?

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/src/SonarAnalyzer.CSharp/RspecStrings.resx (hunk 5 above)
Comment (andrei-epure-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/RspecStrings.resx:12428]: no, it's fine as it is
